core/src/main/scala/ai/h2o/sparkling/backend/H2OChunk.scala

```diff
@@ -75,4 +77,22 @@ private[sparkling] object H2OChunk extends RestCommunication {
       (outputStream: OutputStream) => Compression.compress(conf.externalCommunicationCompression, outputStream)
     insert(endpoint, Paths.CHUNK, conf, addCompression, parameters)
   }
+
+  def putChunkCategoricalDomains(
+      node: NodeDesc,
+      conf: H2OConf,
+      frameName: String,
+      chunkId: Int,
+      domains: Array[Array[String]]): Unit = {
+    val parameters =
+      Map("frame_name" -> frameName, "chunk_id" -> chunkId, "compression" -> conf.externalCommunicationCompression)
+    val endpoint = RestApiUtils.resolveNodeEndpoint(node, conf)
+    val addCompression =
+      (outputStream: OutputStream) => Compression.compress(conf.externalCommunicationCompression, outputStream)
+    withResource(insert(endpoint, Paths.CHUNK_CATEGORICAL_DOMAINS, conf, addCompression, parameters)) { outputStream =>
+      val autoBuffer = new AutoBuffer(outputStream, false)
```

Comment:
I used Kryo since it's Spark dependency and solved problems of Iced serialization on Spark. For communication between Spark and H2O, I would consider also other candidates and do some benchmarks since any other serialization library would be an extra dependency to the extensions assembly.